src/main/java/com/o19s/es/ltr/feature/FeatureValidation.java

```diff
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.o19s.es.ltr.feature;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Simple object to store the parameters needed to validate stored elements:
+ * - The list of template params to replace
+ * - The index to run the query
+ */
+public class FeatureValidation implements Writeable, ToXContentObject {
+    @SuppressWarnings("unchecked")
+    public static final ConstructingObjectParser<FeatureValidation, Void> PARSER = new ConstructingObjectParser<>("feature_validation",
+            (Object[] args) -> new FeatureValidation((String) args[0], (Map<String, Object>) args[1]));
+
+    public static final ParseField INDEX = new ParseField("index");
+
+    public static final ParseField PARAMS = new ParseField("params");
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), INDEX);
+        PARSER.declareField(ConstructingObjectParser.constructorArg(), XContentParser::map,
+                PARAMS, ObjectParser.ValueType.OBJECT);
+    }
+
+    private final String index;
+    private final Map<String, Object> params;
+
+    public FeatureValidation(String index, Map<String, Object> params) {
+        this.index = Objects.requireNonNull(index);
+        this.params = Objects.requireNonNull(params);
+    }
+
+    public FeatureValidation(StreamInput input) throws IOException {
+        this.index = input.readString();
+        this.params = input.readMap();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(index);
+        out.writeMap(params);
+    }
+
+    public String getIndex() {
+        return index;
+    }
+
+    public Map<String, Object> getParams() {
+        return params;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        FeatureValidation that = (FeatureValidation) o;
+
+        if (!index.equals(that.index)) return false;
+        return params.equals(that.params);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = index.hashCode();
```

Comment:
Minor style issue, feel free to ignore.

I find that [`Objects.hash()`](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#hash-java.lang.Object...-) and [`Objects.equals()`](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#equals-java.lang.Object-java.lang.Object-) make for nicer implementations. Note that IntelliJ has a template to generate `equals()` / `hashCode()` using the Objects helper methods.